### ISSUE TITLE
Defining a train method will break the model.eval() call.

### ISSUE THREAD

**Issue by tomsch420** [Zhenye-Na/DA-RNN#21]
Pytorch has a built in method which sets the model into evaluation mode. The method is called torch.nn.Module.eval(). In this method the train method is called with a parameter. Due to the case the your rnn also implements a train method python overwrites the default train method and the code will crash.
How to fix:
I recommend to rename your method into fit() or train_epochs() or something similiear. 

Regardless of some minor issues, your model seems nice.

**Comment by Zhenye-Na** [comment]
Hi @tomsch420 ,

Thank you for your advice, I did not notice this problem at the time I implemented this. Does this affect the repo, since it has been a long time I did not maintain this repo.

Thanks

**Comment by tomsch420** [comment]
I dont know actually if this will affect it anymore. I am using your base implementation and making it more usable for my software architecture right now. I fixed every bug i found but i dont remember them anymore. Maybe I can just create a merge request when i am done turning this into an OOP approach.

**Comment by Zhenye-Na** [comment]
That will be great! I will temporarily close this ticket, but you are always welcome to create a PR.

Thank you!

**Comment by tomsch420** [comment]
So, im done i guess. The network is converging for large amounts of data. But the predictions differ from run to run after loading the weights for the decoder and encoder. Is there some randomness involved that i missed?

**Comment by dan-r95** [comment]
How much do the predictions differ? 
Will you share your OOP approach?
